### PR TITLE
NN-2459 court cases, offences and sentences

### PR DESCRIPTION
**Pull request by JamieDraperUK**
Screenshot 2020-07-01 at 09 33 05

### CODE REVIEW

File: backend/controllers/prisonerProfile/sentenceAndReleaseViewModels/courtCasesViewModel.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+  if (left.months < right.months) return 1
+  if (left.months > right.months) return -1
+
+  return right.days - left.days
+}
+
+module.exports = ({ courtCaseData, sentenceTermsData, offenceHistory }) => {
+  const caseIds = [
+    ...new Set(offenceHistory.filter(offence => offence.primaryResultCode === '1002').map(offence => offence.caseId)),
```
Comment (JamieDraperUK) [inline backend/controllers/prisonerProfile/sentenceAndReleaseViewModels/courtCasesViewModel.js:53]: Going to pull this magic number out into a common file in the next PR.

File: backend/controllers/prisonerProfile/sentenceAndReleaseViewModels/courtCasesViewModel.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    return result
+  }, null)
+
+const groupSentencesBySequence = sentences =>
+  sentences.reduce((result, current) => {
+    const key = current.sentenceSequence
+    const existing = result.find(sentence => sentence.key === key)
+
+    if (existing) {
+      return [{ ...existing, items: [...existing.items, current] }, ...result.filter(entry => entry.key !== key)]
```
Comment (brightonsbox) [inline backend/controllers/prisonerProfile/sentenceAndReleaseViewModels/courtCasesViewModel.js:30]: A JS lesson for me here... when destructuring `existing` and then adding the `items` explicitly, does this overwrite what is brought in by `existing`? _Goes off to investigate...!_

Comment (brightonsbox) [reply backend/controllers/prisonerProfile/sentenceAndReleaseViewModels/courtCasesViewModel.js:30]: ... yes it does 😄

Comment (JamieDraperUK) [reply backend/controllers/prisonerProfile/sentenceAndReleaseViewModels/courtCasesViewModel.js:30]: Ha, you're too fast.

File: backend/controllers/prisonerProfile/sentenceAndReleaseViewModels/courtCasesViewModel.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+  return right.days - left.days
+}
+
+module.exports = ({ courtCaseData, sentenceTermsData, offenceHistory }) => {
+  const caseIds = [
+    ...new Set(offenceHistory.filter(offence => offence.primaryResultCode === '1002').map(offence => offence.caseId)),
+  ]
+
+  return courtCaseData
```
Comment (Nimphal) [inline backend/controllers/prisonerProfile/sentenceAndReleaseViewModels/courtCasesViewModel.js:56]: This is some kind of work of art, honestly

File: backend/controllers/prisonerProfile/sentenceAndReleaseViewModels/courtCasesViewModel.js (hunk 2 above)
Comment (JamieDraperUK) [reply backend/controllers/prisonerProfile/sentenceAndReleaseViewModels/courtCasesViewModel.js:30]: Essentially what's happening here:
`items: [...existing.items, current] }`
Is were are creating a new array with the existing items plus one more item (current)

**Pull request merged**